T&A-0036167-HTML-Quellcode-wird-in-das-Feld-Antwort-Text-übernommen

kunzeus opened the pull request:
Good day,
This fixes the Issue related to: https://mantis.ilias.de/view.php?id=36167

TinyMCE has acces to more HTML tags than the ILIAS Page editor. By switching from Multiple Line Answer to Single Line Answer, the additional TinyMCE HTML tags have not been removed. This fixes the issue without touching TinyMCE, but still allowing for `<em>` and `<strong>` HTML tag which are useable with the ILIAS Page editor.

As always, let me know if this correction can be improved, or if I forgot or didn't know something.

Reviewer:
@kergomard if I remember correctly

kunzeus committed:
[fix] remove unnecesary HTML tags

TinyMCE has acces to more HTML tags than the ILIAS Page editor.
By switching from Multiple Line Answer to Single Line Answer, the additional TinyMCE HTML tags have not been removed. This fixes the issue without touching TinyMCE, but still allowing for <em> and <strong> HTML tag which are useable with the ILIAS Page editor.

jweber-unibe commented:
Hello @kunzeus 
As MC questions of the 'Kprim'-type are also affected, I guess the solution should be extended to those as well. 
Or do I miss something here (esp. from my non developer perspective)?
Kind regards, Jens

kunzeus committed:
[fix] extend fix to kprim inputs

I reused the "setValuesByPost" scheme from single and multiple choice to make the changes of the HTML tag filter persistent.

kunzeus commented:
Hello @jweber-unibe ,

I extended the fix to the 'Kprim'-type 
Note: this is not an ILIAS 8 fix since it uses the now deprecated ilUtil class. I'm starting looking into ILIAS 8 now.

kunzeus committed:
[fix] deprecated code issue

I took a look into ILAIS 8 which still uses deprecated code. Therefore I propose a version which uses php basic function  strip_tags.

kunzeus commented:
The last commit changes the in ILIAS 8 deprecated parts to the PHP strip_tag function. 
It is a proposal? It still needs adaptation to work properly in 8. But I'm unsure of what the correct way would be to fix the issue properly.

dsstrassner commented:
@kergomard 

Could you answer the questions from @kunzeus?

kergomard commented:
Thanks @dsstrassner for the reminder. This got on my backlog and I lost it out of view. Sorry!
Maybe it is easier, if we have a look at this together synchronously?

kergomard commented:
Thank you very much @kunzeus for this PR!

As discussed, I think your PR improves the current situation in the direction of the Mantis-Issue. The trouble is, that the current code does probably not do what it wants to: To sanitize input. I would kindly ask you, to see, if we can actually do what we are intending to do with the ìlAssHTMLQuestionContentPurifier`.

Thank you very much and best,
@kergomard

dsstrassner commented:
Hi @kunzeus, this is ongoing, right?

dsstrassner commented:
TechSquad 11.05.2023: @kunzeus is on this and has asked @kergomard some questions and @kergomard will provide feedback.

kunzeus committed:
revert back to stripSlashesRecursive

also some array() cleanup

kunzeus committed:
protect constant

kergomard commented:
Thank you very much @kunzeus for your PR and your patience with it! I decided on a minimal fix for the time being, so the blame for that decision lies solely on me!
